robot/framesystem/framesystem.go

```diff
@@ -36,6 +36,7 @@ type Service interface {
 	TransformPose(
 		ctx context.Context,
 		pose *referenceframe.PoseInFrame,
+		// NOTE: Why doesn't this take a resource.Name?
```

Comment:
See other comment with tickets referenced.

Framesystems predate resource names existing, and are embedded at a sufficiently low level that we've been able to get away with operating on resource names converted to strings up until now.